src/cms/database/migrations/2025.09.16T00.00.27.park-feature.js

```diff
@@ -0,0 +1,4453 @@
+
+            // get the newName for the matching item
+            featureName = multiItemArea.items.find(
+              (item) => item.featureId === record.featureId
+            )?.newName;
+          }
+        } else {
+          // Check if record.featureId matches a featureId value in singleItemAreas
+          const singleItemArea = singleItemAreas.find(
+            (area) =>
+              area.orcs === record.protectedArea?.orcs &&
+              area.items.some((item) => item.featureId === record.featureId)
+          );
+
+          if (singleItemArea) {
+            // get the orcsAreaNumber for the matching item
+            const orcsAreaNumber = singleItemArea.orcsAreaNumber;
+            if (orcsAreaNumber) {
+              // get the record for park-area that matches the orcsAreaNumber
+              parkArea = await findAreaByOrcsAreaNumber(orcsAreaNumber);
+
+              // this is a special name used for single feature campgrounds
+              featureName = "All sites";
+            }
+          }
+        }
+
+        // get a list of all parkOperationSubAreas and populate the parkSubAreaType
+        const parkOperationSubAreas = await strapi.entityService.findMany(
+          "api::park-operation-sub-area.park-operation-sub-area",
+          {
+            filters: {
+              protectedArea: record.protectedArea.id,
+            },
+            populate: { parkSubAreaType: true },
+          }
+        );
+
+        // Logic for orcsFeatureNumber. We will end up with slightly different values
+        // on different environments, but eventually everything will be restored
+        // with copies of the production database.
+        if (record.protectedArea.id !== previousProtectedAreaId) {
+          counter = 1;
+        } else {
+          counter++;
+        }
+        previousProtectedAreaId = record.protectedArea.id;
+
+        // combine the sharedData with fields that have relationships and special business rules
+        // to create the final data for the new park-feature record
+        const parkFeatureData = {
+          ...sharedData,
+          parkFeature: featureName,
+          publishedAt: new Date().toISOString(),
+          featureId:
+            record.featureId ?? `${record.protectedArea.orcs}_${record.id}`,
+          orcsFeatureNumber: `${record.protectedArea.orcs}-${counter}`,
+          protectedArea: record.protectedArea
+            ? { connect: [{ id: record.protectedArea.id }] }
+            : undefined,
+          site: record.site ? { connect: [{ id: record.site.id }] } : undefined,
+          parkArea: parkArea ? { connect: [{ id: parkArea.id }] } : undefined,
+          parkFeatureType: record.parkSubAreaType
+            ? {
+                connect: [
+                  {
+                    id: featureTypeMap.get(record.parkSubAreaType.subAreaType),
+                  },
+                ],
+              }
+            : undefined,
```

Comment:
Could it be a relation with `parkFeatureType`? and use `parkFeatureType.parkFeatureType` or `parkFeatureType.id` as an id?